app/assets/javascripts/models/beta/projectBoard.js

```diff
@@ -114,16 +142,32 @@ export const moveStory = (
   destinationArray,
   sourceIndex,
   destinationIndex,
+  isSameColumn
 ) => {
   const newSourceArray = [...sourceArray];
   const [removed] = newSourceArray.splice(sourceIndex, 1);
-  const newDestinationArray = [...destinationArray];
-  const filteredArray = newDestinationArray.filter((el, index) => index !== sourceIndex);
-  filteredArray.splice(destinationIndex, 0, removed);
 
-  return [...filteredArray];
+  let newDestinationArray = [...destinationArray];
+  if (isSameColumn) {
+    newDestinationArray = newDestinationArray.filter((el, index) => index !== sourceIndex);
+  }
+  newDestinationArray.splice(destinationIndex, 0, removed);
+  const sortedStories = sortStories(newDestinationArray, destinationIndex, isSameColumn);
+
+  return [...sortedStories];
 };
 
+export const sortStories = (destinationArray, destinationIndex, isSameColumn) => {
+  const newDestinationArray = [...destinationArray];
+  const formatedArray = newDestinationArray.map((item, index) => {
+      if (index >= destinationIndex) {
+        item.newPosition = item.newPosition + 1
+      };
+    return item;
+  });
+  return [...formatedArray];
```

Comment:
Considering the `map` above is iterating over a copy of the `destinationArray` that you already destructure in line 161, do you think we could return that `map` directly instead of destructuring it again? 

```js
export const sortStories = (destinationArray, destinationIndex, isSameColumn) => {
  const newDestinationArray = [...destinationArray];

  return newDestinationArray.map((item, index) => {
    if (index >= destinationIndex) {
      item.newPosition = item.newPosition + 1
    };

    return item;
  });
}
```

Reply:
Hey @wenderjean, sorry I didnt see your comment before or else I would commentend here, but I made a similar comment here: https://github.com/Codeminer42/cm42-central/pull/716#discussion_r576175235 and I believe that is even more complete